src/win32.c

```diff
@@ -1032,4 +1032,34 @@ gchar *win32_get_user_config_dir(void)
 	return g_build_filename(g_get_user_config_dir(), "geany", NULL);
 }
 
+
+/* Retrieve the console codepage
+ * In case GetConsoleCP() returns 0 (i.e. the application doesn't have an own console window
+ * fallback to GetOEMCP(). */
+guint win32_get_console_codepage(void)
+{
+	guint codepage = GetConsoleCP();
+	if (codepage == 0)
+		codepage = GetOEMCP();
+	return codepage;
+}
+
+
+/* Convert a string into the system's default codepage, this is different from the
+ * locale (e.g. default codepage is 850 but locale is CP1252).
+ * This assumes the input string is encoded as UTF-8, otherwise a copy of
+ * the input string is returned. */
+gchar *win32_convert_to_system_codepage(const gchar *str, GError **error)
+{
+	if (g_utf8_validate(str, -1, NULL))
+	{
+		guint codepage_code = win32_get_console_codepage();
+		gchar codepage[8] = { 0 };
+		g_snprintf(codepage, G_N_ELEMENTS(codepage), "%u", codepage_code);
```

Comment:
Also, is the codepage number enough fro `g_convert()`?  shouldn't it rather be something like `CP%u`?  Maybe not, I'm just wondering.